pkg/webservice/handlers.go

```diff
@@ -898,6 +898,56 @@ func getPartitionRules(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getQueueApplicationsByState(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+	vars := httprouter.ParamsFromContext(r.Context())
+	if vars == nil {
+		buildJSONErrorResponse(w, MissingParamsName, http.StatusBadRequest)
+		return
+	}
+
+	partition := vars.ByName("partition")
+	queueName := vars.ByName("queue")
+	appState := strings.ToLower(vars.ByName("state"))
+	status := strings.ToLower(r.URL.Query().Get("status"))
+
+	queueErr := validateQueue(queueName)
+	if queueErr != nil {
+		buildJSONErrorResponse(w, queueErr.Error(), http.StatusBadRequest)
+		return
+	}
+	partitionContext := schedulerContext.GetPartitionWithoutClusterID(partition)
+	if partitionContext == nil {
+		buildJSONErrorResponse(w, PartitionDoesNotExists, http.StatusNotFound)
+		return
+	}
+	queue := partitionContext.GetQueue(queueName)
+	if queue == nil {
+		buildJSONErrorResponse(w, QueueDoesNotExists, http.StatusNotFound)
+		return
+	}
+	if appState != "active" {
+		buildJSONErrorResponse(w, fmt.Sprintf("The provided state is invalid: %s", appState), http.StatusBadRequest)
```

Comment:
Could we align the error message? for example: `"Only following application states are allowed: active"`

Reply:
Good idea. The adjustment has been made.